executors.py

```diff
@@ -178,34 +195,42 @@ def execute(self) -> List[PermutationResult]:
             dimensions.remove(self.dimensions[-1])
 
         # iteration through positions like: n, 0, n-1, 1, n-2, 2, ...
-        for iteration, position in enumerate(chain(*zip(reversed(range(len(dimensions))), range(len(dimensions))))):
+        for iteration, target_position in enumerate(chain(*zip(reversed(range(len(dimensions))), range(len(dimensions))))):
```

Comment:
dimension swaps from original_position to target_position, makes it easier to understand what the swaps are doing